packages/core/src/components/forms/numericInput.tsx

```diff
@@ -0,0 +1,407 @@
+                </div>
+            );
+        } else {
+            const incrementButton = this.renderButton(
+                NumericInput.INCREMENT_KEY, NumericInput.INCREMENT_ICON_NAME, this.handleIncrementButtonClick);
+            const decrementButton = this.renderButton(
+                NumericInput.DECREMENT_KEY, NumericInput.DECREMENT_ICON_NAME, this.handleDecrementButtonClick);
+
+            const buttonGroup = (
+                <div key="button-group" className={classNames(Classes.BUTTON_GROUP, Classes.VERTICAL)}>
+                    {incrementButton}
+                    {decrementButton}
+                </div>
+            );
+
+            const inputClasses = classNames(
+                Classes.NUMERIC_INPUT,
+                Classes.CONTROL_GROUP,
+                {
+                    // because both the <input> and <button>s are nested within
+                    // pt-input-group and pt-button-group divs, respectively, we
+                    // need to keep track of which group has focus in order to
+                    // properly style elements' outlines while focused (we'll
+                    // primarily want to ensure the focused element's outline
+                    // will appear on top of all other elements).
+                    [Classes.NUMERIC_INPUT_BUTTON_GROUP_FOCUSED]: this.state.isButtonGroupFocused,
+                    [Classes.NUMERIC_INPUT_INPUT_GROUP_FOCUSED]: this.state.isInputGroupFocused,
+                },
+                className,
+            );
+            const inputElems = (buttonPosition === Position.LEFT)
+                ? [buttonGroup, inputGroup]
+                : [inputGroup, buttonGroup];
+
+            return (
+                <div className={inputClasses}>
+                    {inputElems}
+                </div>
+            );
+        }
+    }
+
+    public componentWillReceiveProps(nextProps: HTMLInputProps & INumericInputProps) {
+        super.componentWillReceiveProps(nextProps);
+
+        const value = this.getValueOrEmptyValue(nextProps);
+
+        const didMinChange = nextProps.min !== this.props.min;
+        const didMaxChange = nextProps.max !== this.props.max;
+        const didBoundsChange = didMinChange || didMaxChange;
+
+        // if a new min and max were provided that cause the existing value to fall
+        // outside of the new bounds, then clamp the value to the new valid range.
+        if (didBoundsChange) {
+            const { min, max } = nextProps;
+            const adjustedValue = (value !== NumericInput.VALUE_EMPTY)
+                ? this.getAdjustedValue(value, /* delta */ 0, min, max)
+                : NumericInput.VALUE_EMPTY;
+            this.setState({ value: adjustedValue });
+        } else {
+            this.setState({ value });
+        }
+    }
+
+    public componentDidUpdate() {
+        if (this.state.shouldSelectAfterUpdate) {
+            this.inputElement.setSelectionRange(0, this.state.value.length);
```

Comment:
a simple `.select()` may be a more reliable way to select everything.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/select